### PR TITLE
Added support for SPRACINGF3NEO

### PR DESCRIPTION
**Pull request by martinbudden**
@hydra , unfortunately iNav does not support the timer definitions yet, so the `target.c` file is old style. I've made a stab at it, but this is outside my area of expertise, so can you double check (very carefully) the `target.c` file.

### CODE REVIEW

**Comment by digitalentity** [comment]
I'm in favor of merging this and sorting out bugs later if (when) they are found

**Review by digitalentity — APPROVED** [review]

**Comment by martinbudden** [comment]
@digitalentity , this is totally untested at the moment. I'll be able to at least give it some rudimentary testing in the next day or so, so don't merge until then.

**Comment by digitalentity** [comment]
@martinbudden ack

**Comment by digitalentity** [comment]
@hydra were you able to have a brief look on pin/timer mappings?

**Comment by Manawyrm** [comment]
I just compiled the branch and tried to get a basic configuration to work. 
Receiver setup, sensors and UARTs all work absolutly fine.

The timer/pin mapping seems to be relativly wrong, though. 
(http://seriouslypro.com/files/SPRacingF3NEO-Manual-latest.pdf)
iNAV motor 1 is present on the TLM (UART T2) pin! 
iNAV motor 2 is present on the PWM 1 output
iNAV motor 3 is present on the PWM 2 output
iNAV motor 4 is present on the PWM 3 output

the PWM 4 output has no signal present. 

Comparing these two files I can't really spot the problem at first glance, maybe one of you can:
https://github.com/betaflight/betaflight/blob/master/src/main/target/SPRACINGF3NEO/target.c
https://github.com/martinbudden/cleanflight/blob/inav_spracingf3neo/src/main/target/SPRACINGF3NEO/target.c

**Comment by Manawyrm** [comment]
Hm, ok. 
Seeing as UART T2/TLM is PA2 (judging from the F303 datasheet, is there a better way to determine this?), i straight out commented the whole line in the timerHardware struct. 
https://screenshot.tbspace.de/rnhlbtfzkjx.png
This actually works and I can correctly address all 4 motors. I don't know how ESC5 and 6 (and any other pins) behave currently, but I'll figure that out another time.

**Comment by Manawyrm** [comment]
About the NEO OSD/VTX: 
https://tbspace.de/content/images/PICT0044.JPG
https://tbspace.de/content/images/PICT0046.JPG

The video signal is present on the board and the OSD is correctly controlled, but the VTX unfortunatly doesn't get initialized at all. Is it possible that iNAV has no VTX support whatsoever? Should we port the VTX support from Cleanflight over? (there also was no option to enable VTX in the configuration-tab)

EDIT: I just saw that https://github.com/iNavFlight/inav/pull/1576/ already ports the VTX support over. 
imho the VTX suppport should be taken out of this PR, and then commited together with the SmartAudio/VTX support.

**Comment by Manawyrm** [comment]
Success! 
I have merged the VTX support, as well as the RTC6705 support into iNAV and I can now use the full NEO VTX stack (with OSD control) in iNAV.

I would suggest this pull request gets merged quickly, so we can then fix the last remaining issues with the VTX  pull requests and commit it afterwards.

File: src/main/target/SPRACINGF3NEO/target.c [hunk 1]
```diff
@@ -0,0 +1,119 @@
+/*
```
Comment (digitalentity) [inline src/main/target/SPRACINGF3NEO/target.c:1]: Please align timer definitions with new timer framework - see other targets for how it's done now.

File: src/main/target/SPRACINGF3NEO/target.h [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+#define MPU6500_SPI_INSTANCE    SPI1
+
+#define DEFAULT_VOLTAGE_METER_SOURCE VOLTAGE_METER_ADC
+#define DEFAULT_CURRENT_METER_SOURCE CURRENT_METER_ADC
+
+#define USE_ADC
+#define ADC_INSTANCE            ADC1
+#define VBAT_ADC_PIN            PC1
+#define CURRENT_METER_ADC_PIN   PC2
+#define RSSI_ADC_PIN            PC0
```
Comment (digitalentity) [inline src/main/target/SPRACINGF3NEO/target.h]: Please align with latest ADC channel mapping code.

File: src/main/target/SPRACINGF3NEO/target.h [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+// Note, this is the same DMA channel as UART1_RX. Luckily we don't use DMA for USART Rx.
+#define SDCARD_DMA_CHANNEL_TX               DMA1_Channel5
+#define SDCARD_DMA_CHANNEL_TX_COMPLETE_FLAG DMA1_FLAG_TC5
+
+#define MPU6500_CS_PIN          SPI1_NSS_PIN
+#define MPU6500_SPI_INSTANCE    SPI1
+
+#define DEFAULT_VOLTAGE_METER_SOURCE VOLTAGE_METER_ADC
+#define DEFAULT_CURRENT_METER_SOURCE CURRENT_METER_ADC
```
Comment (digitalentity) [inline src/main/target/SPRACINGF3NEO/target.h]: These are not used in INAV

File: src/main/target/SPRACINGF3NEO/target.h [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+#define USE_LED_STRIP_ON_DMA1_CHANNEL2
+#define WS2811_PIN                      PA8
+#define WS2811_TIMER                    TIM1
+#define WS2811_DMA_CHANNEL              DMA1_Channel2
+#define WS2811_IRQ                      DMA1_Channel2_IRQn
+#define WS2811_DMA_TC_FLAG              DMA1_FLAG_TC2
+#define WS2811_DMA_HANDLER_IDENTIFER    DMA1_CH2_HANDLER
+#define WS2811_TIMER_GPIO_AF            GPIO_AF_6
+
+#define TRANSPONDER
```
Comment (digitalentity) [inline src/main/target/SPRACINGF3NEO/target.h]: Transponder is not supported

**Review by digitalentity — COMMENTED** [review]
Please align timer and ADC code with recent changes.

**Comment by digitalentity** [comment]
@martinbudden can you aligh this with new timer/ADC definitions so we can merge it?

**Comment by martinbudden** [comment]
@digitalentity , I'm not really very familiar with the new timer definitions - can you make that change?

**Comment by digitalentity** [comment]
@martinbudden no problem, will do

**Comment by digitalentity** [comment]
Will replace this PR with a new one

**Pull request closed without merging**